csharp/src/Apache.Arrow/Memory/LargeMemoryAllocator.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Apache.Arrow.Memory
+{
+    internal abstract class LargeMemoryAllocator
```

Comment:
I think rather than a separate class, we should add methods for large memory allocation to the existing `MemoryAllocator` class.